plugins/scaffolder-backend-module-bitbucket-cloud/src/actions/helpers.ts

```diff
@@ -21,22 +21,22 @@ export const getBitbucketClient = (config: {
   username?: string;
   appPassword?: string;
 }) => {
-  if (config.username && config.appPassword) {
+  if (config.token) {
     return new Bitbucket({
       auth: {
-        username: config.username,
-        password: config.appPassword,
+        token: config.token,
       },
     });
-  } else if (config.token) {
+  } else if (config.username && config.appPassword) {
     return new Bitbucket({
       auth: {
-        token: config.token,
+        username: config.username,
+        password: config.appPassword,
       },
     });
   }
   throw new Error(
-    `Authorization has not been provided for Bitbucket Cloud. Please add either username + appPassword to the Integrations config or a user login auth token`,
+    `Authorization has not been provided for Bitbucket Cloud. Please add either provide a username + token or  username + appPassword to the Integrations config`,
```

Comment:
I actually don't think we need these changes to `getBitbucketClient`, they work with a token already in the library being used: https://github.com/MunifTanjim/node-bitbucket.